pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -255,6 +265,42 @@ type TriggerParameterSource struct {
 	Value *string `json:"value,omitempty" protobuf:"bytes,3,opt,name=value"`
 }
 
+// TriggerPolicy dictates the policy for the trigger retries
+type TriggerPolicy struct {
+	// Backoff before checking resource state
+	Backoff Backoff `json:"backoff" protobuf:"bytes,1,opt,name=backoff"`
+
+	// Criteria to check resource state
+	Criteria *TriggerPolicyCriteria `json:"criteria" protobuf:"bytes,2,opt,name=criteria"`
+
+	// ErrorOnBackoffTimeout determines whether sensor should transition to error state if the backoff times out and yet the resource neither transitioned into success or failure.
+	ErrorOnBackoffTimeout bool `json:"errorOnBackoffTimeout" protobuf:"bytes,3,opt,name=errorOnBackoffTimeout"`
+}
+
+// Backoff for an operation
+type Backoff struct {
+	// Duration is the duration in nanoseconds
+	Duration time.Duration `json:"duration" protobuf:"bytes,1,opt,name=duration"`
+
+	// Duration is multiplied by factor each iteration
+	Factor float64 `json:"factor" protobuf:"bytes,2,opt,name=factor"`
+
+	// The amount of jitter applied each iteration
+	Jitter float64 `json:"jitter" protobuf:"bytes,3,opt,name=jitter"`
+
+	// Exit with error after this many steps
+	Steps int `json:"steps" protobuf:"bytes,4,opt,name=steps"`
+}
+
+// TriggerPolicyCriteria defines the criteria to decide if a resource is in success or failure state.
+type TriggerPolicyCriteria struct {
```

Comment:
how are these used? it seems like both are labels? maybe rename to to include that fact? also, what is the use case for having this?

Reply:
Basically, the labels defined in this struct are used to figure out if the resource has reached success state or it failed. We are using labels because the sensor triggers an unstructured object, so it is not possible to simply get the particular field that is supposed to reflect the state of the resource as you would normally do in a structured K8s object. In case of workflows, the controller sets the labels like `workflows.argoproj.io/phase: Succeeded`  or `workflows.argoproj.io/phase: Failed` to mark workflow as successful or failed.

I can rename the struct to `TriggerStateLabels`.  

The use case is that if the sensor has multiple triggers to fire but the execution of each trigger must depend on the success or failure of the previous trigger.